internal/handlers/handlers.go

```diff
@@ -0,0 +1,284 @@
+package handlers
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/gofiber/fiber/v2"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/readpref"
+
+	"github.com/threefoldtech/tf-kyc-verifier/internal/build"
+	"github.com/threefoldtech/tf-kyc-verifier/internal/configs"
+	"github.com/threefoldtech/tf-kyc-verifier/internal/errors"
+	"github.com/threefoldtech/tf-kyc-verifier/internal/logger"
+	"github.com/threefoldtech/tf-kyc-verifier/internal/models"
+	"github.com/threefoldtech/tf-kyc-verifier/internal/responses"
+	"github.com/threefoldtech/tf-kyc-verifier/internal/services"
+)
+
+type Handler struct {
+	kycService services.KYCService
+	config     *configs.Config
+	logger     logger.Logger
+}
+
+func NewHandler(kycService services.KYCService, config *configs.Config, logger logger.Logger) *Handler {
+	return &Handler{kycService: kycService, config: config, logger: logger}
+}
+
+// @Summary		Get or Generate iDenfy Verification Token
+// @Description	Returns a token for a client
+// @Tags			Token
+// @Accept			json
+// @Produce		json
+// @Param			X-Client-ID	header		string	true	"TFChain SS58Address"								minlength(48)	maxlength(48)
+// @Param			X-Challenge	header		string	true	"hex-encoded message `{api-domain}:{timestamp}`"
+// @Param			X-Signature	header		string	true	"hex-encoded sr25519|ed25519 signature"				minlength(128)	maxlength(128)
+// @Success		200			{object}	responses.TokenResponse "Existing token retrieved"
+// @Success		201			{object}	responses.TokenResponse "New token created"
+// @Failure		400			{object}	responses.ErrorResponse
+// @Failure		401			{object}	responses.ErrorResponse
+// @Failure		402			{object}	responses.ErrorResponse
+// @Failure		409			{object}	responses.ErrorResponse
+// @Failure		500			{object}	responses.ErrorResponse
+// @Router			/api/v1/token [post]
+func (h *Handler) GetorCreateVerificationToken() fiber.Handler {
+	return func(c *fiber.Ctx) error {
+		clientID := c.Get("X-Client-ID")
+		ctx, cancel := context.WithTimeout(c.Context(), 5*time.Second)
+		defer cancel()
+		token, isNewToken, err := h.kycService.GetorCreateVerificationToken(ctx, clientID)
+		if err != nil {
+			return HandleError(c, err)
+		}
+		response := responses.NewTokenResponseWithStatus(token, isNewToken)
+		if isNewToken {
+			return c.Status(fiber.StatusCreated).JSON(fiber.Map{"result": response})
```

Comment:
Why are we returning nested responses? you already created the response struct in 57? why embedding that in a map?

Reply:
I see that you always return result/error, I suggest that you create some sort of helpers for success and error that you pass the object to and it does the fiber.Map

Reply:
Done